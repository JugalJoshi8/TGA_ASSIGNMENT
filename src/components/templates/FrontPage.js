import { useState } from "react";
import Head from "next/head";
import { withRouter } from "next/router";

import Link from "next/link";
import fetch from "isomorphic-unfetch";
import NewsItem from "../molecules/NewsItem";

export const FrontPage = ({ router, hits }) => {
  console.log(router);
  const currentPage = router.query.page || 0;
  const nextPageLink = `/news?page=${parseInt(currentPage) + 1}`;
  return (
    <div className="container">
      <Head>
        <title>Hacker news</title>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="contains all the latest hacking related news"
        />
      </Head>
      <main>
        <ul className="news-list">
          {hits.map((item) => (
            <NewsItem key={item.objectID} item={item} />
          ))}
        </ul>
        <Link
          href={{ pathname: "/", query: { page: parseInt(currentPage) + 1 } }}
          as={nextPageLink}
        >
          <a className="load-more" href={nextPageLink}>
            More
          </a>
        </Link>
      </main>

      <style jsx>{`
        ul {
          margin-bottom: 20px;
        }

        .load-more {
          cursor: pointer;
          margin-bottom: 20px;
          display: block;
        }
      `}</style>
    </div>
  );
};

FrontPage.getInitialProps = async (ctx) => {
  console.log("ctx", ctx.query);
  const res = await fetch(
    `https://hn.algolia.com/api/v1/search?page=${ctx.query.page || 0}`
  );
  const json = await res.json();
  return { hits: json.hits };
};

export default withRouter(FrontPage);
