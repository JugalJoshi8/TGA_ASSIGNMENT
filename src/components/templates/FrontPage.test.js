import React from "react";
import { shallow, mount } from "enzyme";
import FrontPage from "./FrontPage";

const items = [
  {
    num_comments: 110,
    upvotes: 10,
    objectID: 1234,
    url: "www.test.com",
  },
];

let wrapped = shallow(<FrontPage hits={items} />);

describe("NewsItem", () => {
  it("should render the FrontPage Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
  it("renders the heading title", () => {
    wrapped = mount(<FrontPage hits={items} />);
    expect(wrapped.find(".comments").text()).toEqual("110");
  });
});
