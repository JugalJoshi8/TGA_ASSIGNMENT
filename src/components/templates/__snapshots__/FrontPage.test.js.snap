// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem should render the FrontPage Component correctly 1`] = `ShallowWrapper {}`;
